toolchain/lex/lex.cpp

```diff
@@ -1179,17 +1175,43 @@ auto Lexer::LexWordAsTypeLiteralToken(llvm::StringRef word, int32_t byte_offset)
       return LexResult::NoMatch();
   };
 
+  // No leading zeros allowed.
+  if ('1' > word[1] || word[1] > '9') {
+    return LexResult::NoMatch();
+  }
+
   llvm::StringRef suffix = word.substr(1);
-  if (!CanLexInt(emitter_, suffix)) {
```

Comment:
Looking at other callers for CanLexInt, should string literals be sharing the logic? i.e., should CanLexInt be factored into just numeric literals?

Reply:
Maybe? I'm happy to look at those next?

Reply:
SG

Reply:
Actually, thinking about this further... maybe you should look at it here, because you're taking a shared function and implementing a specialized version, rather than continuing sharing? i.e., to me this is a regression from a pre-existing shared code state.

Reply:
As discussed, happy to look at this, but would prefer in a follow-up.